torch/distributed/algorithms/model_averaging/averagers.py

```diff
@@ -109,5 +112,5 @@ def average_parameters(self, params):
             self.step >= self.warmup_steps
             and (self.step - self.warmup_steps) % self.period == 0
         ):
-            utils.average_parameters(iter(params), self.process_group)
+            utils.average_parameters(params, self.process_group)
```

Comment:
Does this make any difference in numerical results, or is it mainly an API change?

Reply:
iter a torch.nn.parameter will make difference.
actually @wayi1 has fix the bug in the following cr.
https://github.com/pytorch/pytorch/pull/74737/files
i will optimize the PeriodicModelAverager() api to supoort model.parameters() input and optimer.param_groups input.
And one unit test to make sure model average with optimer.param_groups input  get a correct result